src/containers/qvector.c

```diff
@@ -830,21 +830,16 @@ void qvector_reverse(qvector_t *vector) {
 
     int i;
     int j;
-    void *tmp = malloc(vector->objsize);
-    if (tmp == NULL) {
-        errno = ENOMEM;
-        return;
-    }
-
     for (i = 0, j = vector->num - 1; i < j; i++, j--) {
-        void *data1 = (unsigned char *)vector->data + i * vector->objsize;
-        void *data2 = (unsigned char *)vector->data + j * vector->objsize;
+        unsigned char *data1 = (unsigned char *)vector->data + i * vector->objsize;
+        unsigned char *data2 = (unsigned char *)vector->data + j * vector->objsize;
 
-        memcpy(tmp, data1, vector->objsize);
-        memcpy(data1, data2, vector->objsize);
-        memcpy(data2, tmp, vector->objsize);
+        for (int k = 0; k < vector->objsize; k++) {
```

Comment:
I think the behavior of the qvector_reverse() is to reverse the order of the elements, not the byte order inside the elements.

Reply:
My bad. Plz ignore the above comment. I misread.
But performance-wise, how will it compare?

Reply:
don't need call malloc